src/test/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherITest.java

```diff
@@ -1,393 +0,0 @@
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-import org.jvnet.hudson.test.Issue;
-import org.jvnet.hudson.test.LoggerRule;
-import org.mockito.MockedStatic;
-
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.DeserializationFeature;
-import com.fasterxml.jackson.databind.InjectableValues;
-import com.fasterxml.jackson.databind.MapperFeature;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.ObjectReader;
-import com.fasterxml.jackson.databind.PropertyNamingStrategy;
-import com.fasterxml.jackson.databind.introspect.VisibilityChecker;
-import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
-import com.github.tomakehurst.wiremock.junit.WireMockRule;
-
-import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
-
-import org.kohsuke.github.GHCheckRun;
-import org.kohsuke.github.GHCheckRunBuilder;
-import org.kohsuke.github.GHRepository;
-import org.kohsuke.github.GitHub;
-import org.jenkinsci.plugins.displayurlapi.ClassicDisplayURLProvider;
-import org.jenkinsci.plugins.github_branch_source.Connector;
-import org.jenkinsci.plugins.github_branch_source.GitHubAppCredentials;
-import org.jenkinsci.plugins.github_branch_source.GitHubSCMSource;
-import org.jenkinsci.plugins.github_branch_source.PullRequestSCMRevision;
-import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
-import org.jenkinsci.plugins.workflow.job.WorkflowJob;
-import hudson.model.FreeStyleProject;
-import hudson.model.Job;
-import hudson.model.Queue;
-import hudson.model.Run;
-import hudson.util.Secret;
-import jenkins.model.ParameterizedJobMixIn;
-import jenkins.scm.api.SCMHead;
-
-import io.jenkins.plugins.checks.api.ChecksAction;
-import io.jenkins.plugins.checks.api.ChecksAnnotation.ChecksAnnotationBuilder;
-import io.jenkins.plugins.checks.api.ChecksAnnotation.ChecksAnnotationLevel;
-import io.jenkins.plugins.checks.api.ChecksConclusion;
-import io.jenkins.plugins.checks.api.ChecksDetails;
-import io.jenkins.plugins.checks.api.ChecksDetails.ChecksDetailsBuilder;
-import io.jenkins.plugins.checks.api.ChecksImage;
-import io.jenkins.plugins.checks.api.ChecksOutput.ChecksOutputBuilder;
-import io.jenkins.plugins.checks.api.ChecksStatus;
-import io.jenkins.plugins.util.IntegrationTestWithJenkinsPerTest;
-import io.jenkins.plugins.util.PluginLogger;
-
-import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.*;
-import static io.jenkins.plugins.checks.github.assertions.Assertions.*;
-import static org.mockito.Mockito.*;
-
-/**
- * Tests if the {@link GitHubChecksPublisher} actually sends out the requests to GitHub in order to publish the check
- * runs.
- */
-@RunWith(Parameterized.class)
-@SuppressWarnings({"PMD.ExcessiveImports", "checkstyle:ClassDataAbstractionCoupling", "rawtypes", "checkstyle:ClassFanOutComplexity", "checkstyle:JavaNCSS"})
-public class GitHubChecksPublisherITest extends IntegrationTestWithJenkinsPerTest {
```

Comment:
This `IntegrationTestWithJenkinsPerTest` is the problem. Should be rewritten to use just stuff provided by `jenkins-test-harness`.